teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -224,24 +328,349 @@ The entrypointer image should also be able to see that this flag is set and acco
+        CERTIFICATE-----\n","type":1},{"key":"bar","value":"world","type":1},{"key":"bar.sig","value":"MEUCIQDOtg+aEP1FCr6/FsHX+bY1d5abSQn2kTiUMg4Uic2lVQIgTVF5bbT/O77VxESSMtQlpBreMyw2GmKX2hYJlaOEH1M=","type":1},{"key":"foo","value":"hello","type":1},{"key":"foo.sig","value":"MEQCIBr+k0i7SRSyb4h96vQE9hhxBZiZb/2PXQqReOKJDl/rAiBrjgSsalwOvN0zgQay0xQ7PRbm5YSmI8tvKseLR8Ryww==","type":1}]'
+```
+
+Parsed, the fields woud be:
+```
+ ∙ RESULT_MANIFEST       foo,bar
+ ∙ RESULT_MANIFEST.sig   MEQCIB4grfqBkcsGuVyoQd9KUVzNZaFGN6jQOKK90p5HWHqeAiB7yZerDA+YE3Af/ALG43DQzygiBpKhTt8gzWGmpvXJFw==
+ ∙ SVID                  -----BEGIN CERTIFICATE-----
+MIICCjCCAbCgAwIBAgIRALH94zAZZXdtPg97O5vG5M0wCgYIKoZIzj0EAwIwHjEL
+MAkGA1UEBhMCVVMxDzANBgNVBAoTBlNQSUZGRTAeFw0yMjAzMTQxNTUzNTlaFw0y
+MjAzMTQxNjU0MDlaMB0xCzAJBgNVBAYTAlVTMQ4wDAYDVQQKEwVTUElSRTBZMBMG
+ByqGSM49AgEGCCqGSM49AwEHA0IABPLzFTDY0RDpjKb+eZCIWgUw9DViu8/pM8q7
+HMTKCzlyGqhaU80sASZfpkZvmi72w+gLszzwVI1ZNU5e7aCzbtSjgc8wgcwwDgYD
+VR0PAQH/BAQDAgOoMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjAMBgNV
+HRMBAf8EAjAAMB0GA1UdDgQWBBSsUvspy+/Dl24pA1f+JuNVJrjgmTAfBgNVHSME
+GDAWgBSOMyOHnyLLGxPSD9RRFL+Yhm/6qzBNBgNVHREERjBEhkJzcGlmZmU6Ly9l
+eGFtcGxlLm9yZy9ucy9kZWZhdWx0L3Rhc2tydW4vbm9uLWZhbHNpZmlhYmxlLXBy
+b3ZlbmFuY2UwCgYIKoZIzj0EAwIDSAAwRQIhAM4/bPAH9dyhBEj3DbwtJKMyEI56
+4DVrP97ps9QYQb23AiBiXWrQkvRYl0h4CX0lveND2yfqLrGdVL405O5NzCcUrA==
+-----END CERTIFICATE-----
+ ∙ bar       world
+ ∙ bar.sig   MEUCIQDOtg+aEP1FCr6/FsHX+bY1d5abSQn2kTiUMg4Uic2lVQIgTVF5bbT/O77VxESSMtQlpBreMyw2GmKX2hYJlaOEH1M=
+ ∙ foo       hello
+ ∙ foo.sig   MEQCIBr+k0i7SRSyb4h96vQE9hhxBZiZb/2PXQqReOKJDl/rAiBrjgSsalwOvN0zgQay0xQ7PRbm5YSmI8tvKseLR8Ryww==
+```
+
+
+However, the verification material be removed from the results as part of the TaskRun status:
+```console
+$ tkn tr describe non-falsifiable-provenance
+Name:              non-falsifiable-provenance
+Namespace:         default
+Service Account:   default
+Timeout:           1m0s
+Labels:
+ app.kubernetes.io/managed-by=tekton-pipelines
+
+🌡️  Status
+
+STARTED          DURATION     STATUS
+38 seconds ago   36 seconds   Succeeded
+
+📝 Results
+
+ NAME        VALUE
+ ∙ bar       world
+ ∙ foo       hello
+
+🦶 Steps
+
+ NAME                STATUS
+ ∙ non-falsifiable   Completed
+```
+
+An indication that verification has taken place will be as a condition of the TaskRun status:
+```
+  conditions:
```

Comment:
please add a condition in case of failed verification? which all scenarios can result in such failure and what would condition look like?

Reply:
Right now, the status would be "False" if it fails with the reason encoded. Would that be sufficient? Would like to hear your thoughts.

`status`|`reason`|`completionTime` is set|Description
:-------|:-------|:---------------------:|--------------:
True|TaskRunResultsVerified|Yes|The `TaskRun` results have been verified through validation of its signatures
False|TaskRunResultsVerificationFailed|Yes|The `TaskRun` results' signatures failed to verify
Unknown|AwaitingTaskRunResults|No|Waiting upon `TaskRun` results and signatures to verify

Reply:
This follows the current format of the other conditions within pipeline. So for consistency this is how the condition for spire verification was done.